coil/src/main/java/dev/chrisbanes/accompanist/coil/Coil.kt

```diff
@@ -0,0 +1,264 @@
+    alignment: Alignment = Alignment.Center,
+    contentScale: ContentScale = ContentScale.Fit,
+    crossfadeDuration: Int = 1000,
+    modifier: Modifier = Modifier
+) {
+    WithConstraints(modifier) { constraints, _ ->
+        var imgLoadState by stateFor(data) { ImageLoadState.Empty }
+
+        val transitionDef = remember(crossfadeDuration) {
+            transitionDefinition {
+                state(ImageLoadState.Empty) {
+                    this[alpha] = 0f
+                    this[brightness] = 0.8f
+                    this[saturation] = 0f
+                }
+                state(ImageLoadState.Loaded) {
+                    this[alpha] = 1f
+                    this[brightness] = 1f
+                    this[saturation] = 1f
+                }
+
+                transition {
+                    alpha using tween {
+                        duration = crossfadeDuration / 2
+                    }
+                    brightness using tween {
+                        duration = (crossfadeDuration * 0.75f).roundToInt()
+                    }
+                    saturation using tween {
+                        duration = crossfadeDuration
+                    }
+                }
+            }
+        }
+
+        val requestWidth = when {
+            constraints.hasFixedWidth -> constraints.maxWidth
+            constraints.hasBoundedWidth -> constraints.maxWidth
+            else -> constraints.minWidth
+        }
+
+        val requestHeight = when {
+            constraints.hasFixedHeight -> constraints.maxHeight
+            constraints.hasBoundedHeight -> constraints.maxHeight
+            else -> constraints.minHeight
+        }
+
+        val image = loadImage(data, requestWidth.value, requestHeight.value, transformations) {
+            // Once loaded, update the load state
+            imgLoadState = ImageLoadState.Loaded
+        }
+
+        Transition(definition = transitionDef, toState = imgLoadState) { transitionState ->
+            if (image != null) {
+                // Create and update the ImageLoadingColorMatrix from the transition state
+                val matrix = remember(image) { ImageLoadingColorMatrix() }
+                matrix.saturationFraction = transitionState[saturation]
+                matrix.alphaFraction = transitionState[alpha]
+                matrix.brightnessFraction = transitionState[brightness]
+
+                // Unfortunately ColorMatrixColorFilter is not mutable so we have to create a new
+                // instance every time
+                val cf = ColorMatrixColorFilter(matrix)
+                Image(
+                        painter = AndroidColorMatrixImagePainter(image, cf),
+                        contentScale = contentScale,
+                        alignment = alignment
+                )
+            }
+        }
+    }
+}
+
+/**
+ * A simple composable which loads an image using [Coil] into a [Box]
+ */
+@Composable
+fun NetworkImage(
+    data: Any,
```

Comment:
Since this takes `data`, it could technically work with any supported Coil [data type](https://coil-kt.github.io/coil/getting_started/#supported-data-types). What do you think about renaming it to `CoilImage` or just make it an overload of `Image`?

You could also make `data` a `Uri` or `HttpUrl`, however that's more restrictive.

Reply:
> What do you think about renaming it to CoilImage

Yep, done! I personally like the `data: Any` thing, because I use mapped types extensively.